### PR TITLE
Adds hair color gradients

### PR DESCRIPTION
**Pull request by ivanmixo**
## About The Pull Request

For all your snowflake needs. Felinids should apply gradients with caution.

Also includes a minor renaming in `preferences.dm`, `hair` => `hair_color` and minor spacing in `preferences_savefile`.

image

Ports: 
https://github.com/tgstation/tgstation/pull/59435 - without the dyespray and quirk, I made it a pref instead.
https://github.com/tgstation/tgstation/pull/59545

## Why It's Good For The Game

Allows people to have more unique hair colors.

## Changelog
:cl:
add: Added hair color gradients
/:cl:

### CODE REVIEW

**Comment by AnCopper** [comment]
Byakuren.

**Comment by bluezorua** [comment]
dont we already have a "hair dye painter" could give it this as an alt feature

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 1]
```diff
@@ -7,24 +7,24 @@
...
+	///Hair colour
 	var/hair_color = "000"
+	///Hair style
 	var/hair_style = "Bald"
-
-	//Facial hair colour and style
+	///Colour used for the hair gradient.
+	var/gradient_color = "000"
+	///Style used for the hair gradient.
+	var/gradient_style
```
Comment (Archanial) [inline code/modules/mob/living/carbon/human/human_defines.dm:17]: Why do they need to be defined on both humans and species?

Comment (ivanmixo) [reply code/modules/mob/living/carbon/human/human_defines.dm:17]: Moved it off

**Comment by Isy232** [comment]
The more "anime protagonist" you can make your character the better I will feel about killing them or giving them the shitcurity treatment.

Good PR

**Comment by ivanmixo** [comment]
Your gradient doesn't have to be very obvious. There are more subtle ones too to make your hair a bit better looking.

**Comment by ivanmixo** [comment]
no

**Comment by Kapu1178** [comment]
This is my formal seal of carbon freeze approval.

MERRGEEEEEE

**Comment by ike709** [comment]
🔑 

A lot of this doesn't look good IMO.

**Comment by PowerfulBacon** [comment]
0i THINK THAT THE GRADIENT WITH JUST A FADE LOOKS ALRIGHT, BUT THE SPIKEY ONE LOOKS TERRIBLE.

**Comment by AnCopper** [comment]
> 0i THINK THAT THE GRADIENT WITH JUST A FADE LOOKS ALRIGHT, BUT THE SPIKEY ONE LOOKS TERRIBLE.

No need to scream bro.

File: code/modules/mob/living/carbon/human/human_defines.dm [hunk 2]
```diff
@@ -7,24 +7,24 @@
 	buckle_lying = FALSE
 	mob_biotypes = list(MOB_ORGANIC, MOB_HUMANOID)
 	blocks_emissive = EMISSIVE_BLOCK_UNIQUE
-	//Hair colour and style
+	///Hair colour
```
Comment (Archanial) [inline code/modules/mob/living/carbon/human/human_defines.dm]:
```suggestion
	///Hair color
```

**Review by Archanial — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_defines.dm (hunk 2 above)
Comment (ivanmixo) [reply code/modules/mob/living/carbon/human/human_defines.dm]: bri*ish

**Review by Archanial — DISMISSED** [review]

**Comment by Archanial** [comment]
I think those are fine but I will leave decision to other maints

**Comment by BrumalLake** [comment]
i love this pr this will make plasmamen so much better with the amount of customization this allows

**Comment by ivanmixo** [comment]
Code plasmaman helmet gradients when??

**Review by Archanial — APPROVED** [review]

**Pull request merged**